### PR TITLE
Include time to load by default

### LINKED ISSUE #3: feature request: measure Time To Load (TTL)
the result of `@time @eval using PackageName` after precompilation has been ensured (probably can not be part of a BenchmarkGroup)

### PR DESCRIPTION
**Pull request by MilesCranmer**
Fixes #3.

@Krastanov could you give a quick review? Thanks!

(Also - I need to modify the unit tests)

### CODE REVIEW

**Comment by Krastanov** [comment]
Thank you for adding this! It looks good to me! The obvious drawback is that it runs a single sample, but the only way to run more samples is to have launch more temporary julia processes. Not sure how worthwhile that is (maybe it is necessary if in the future there is a benchmark for precompile time?)

**Comment by MilesCranmer** [comment]
What do you think about if we were to include a flag `--nsamples-load-time`, which is default 1 (to keep the default benchmark quick), but could be increased? That would control the number of times Julia is started, in order to record the evaluation time.

**Comment by Krastanov** [comment]
Is that doable without very significant refactor!? If so, yeah, I would be eager to use such an option.

**Comment by MilesCranmer** [comment]
Yeah, should be easy!

**Comment by MilesCranmer** [comment]
Done. For example:

```bash
benchpkg AirspeedVelocity -r master --nsamples-load-time=5
```

Then,

```bash
benchpkgtable AirspeedVelocity -r master
```

the table now has an extra entry:

|                   | master          |
|:------------------|:---------------:|
| main/random_sleep | 0.506 ± 0.22 s  |
| time_to_load      | 2.18 ± 0.0096 s |

What do you think? I'll set the PR as ready for review if you are up for it :)

**Comment by MilesCranmer** [comment]
It's also not as slow as I thought it would be. So perhaps we could set the default to, e.g., 5?

File: src/Utils.jl [hunk 1]
```diff
@@ -173,9 +190,29 @@ function _benchmark(
...
+            io = IOBuffer()
+            run(cmd(io))
+            push!(load_times, parse(Float64, String(take!(io))))
+        end
+        @info "    Done time-to-load tests."
+    end
+    # Write to results file:
+    open(results_filename, "w") do io
+        JSON3.write(io, results)
+    end
```
Comment (MilesCranmer) [inline src/Utils.jl:215]: (Here is where the load time is recorded)

File: README.md [hunk 2]
```diff
@@ -146,6 +146,8 @@ Benchmark a package over a set of revisions.
 - `--path <arg>`: Path of the package.
 - `--bench-on <arg>`: If the script is not set, this specifies the revision at which
   to download `benchmark/benchmarks.jl` from the package.
+- `--nsamples-load-time <arg>`: Number of samples to take when measuring load time of
+    the package (default: 1). (This means starting a Julia process for each sample.)
```
Comment (Krastanov) [inline README.md:150]: just double checking, there are a few places where it defaults to 5 below. Is this docstring outdated?

**Comment by Krastanov** [comment]
I read through the additions. I am not yet familiar with this code base, but nothing bad popped up at me.

**Comment by Krastanov** [comment]
Thanks for implementing this. It would be great to use it when merged

File: README.md (hunk 2 above)
Comment (MilesCranmer) [reply README.md:150]: Yes, good catch!

**Comment by MilesCranmer** [comment]
Thanks, merging!

**Pull request merged**